sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/models/ChatMessageDeletedEvent.java

```diff
@@ -12,28 +14,33 @@
      * The timestamp when the message was deleted. The timestamp is in ISO8601 format:
      * `yyyy-MM-ddTHH:mm:ssZ`.
      */
+    @JsonProperty(value = "deletedOn")
     private String deletedOn;
 
     /**
      * The Id of the message. This Id is server generated.
      */
+    @JsonProperty(value = "id")
```

Comment:
In case event types are output only (from service -> user) do we need `set*` methods anymore?

Reply:
We need the setters when parsing this object from json from trouter.
The field names are different between json from trouter and what we expose to customer. There's a mapping in implementation/signaling/TrouterUtils.java

Reply:
When we talked about using `JacksonSerder`, I thought it is to replace the parsing logic in `TrouterUtil` from `JsonObject` to `Jackson`, but it seems we use it only to simplify the logging experience in apps.

But now we can remove those "public setters (not usable for customers)," Jackson helps here; it does not require setter methods to set value for properties with "JsonProperty" annotation.

Special care is needed to populate those properties with the type "CommunicationIdentifier" in `ChatUserEvent`; since the wire representation is different, Jackson cannot set those properties. We can use an accessor pattern to solve that.

Here is a sample impl showing removal of unused setters from our public API and the accessor pattern in impl level; I did it only for `Deleted` and `Edited` event types but the concept is same across event models.

https://github.com/anuchandy/azure-sdk-for-android-1/commit/d7a280293714438a9c02a5a6166884b89382dfb5

Reply:
If we want to use `JacksonSerder` for parsing `JsonObject`, it means we need to put quite a lot extra private properties in event classes. For example, for `ChatMessageReceivedEvent`, we need to add `groupId`, `messageId`, `originalArrivalTime` and `deletetime` with 4 package private set functions. Is there a way to prevent those 4 extra properties to be printed if customer use `JacksonSerder` to print log?

Reply:
If we want to omit few properties when serializing then Jackson's WRITE_ONLY annotation can be used, like [this](https://github.com/Azure/azure-sdk-for-android/blob/e4a006d662ca2825cf4fe991896c8d3f962574b3/sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/models/CommunicationError.java#L29). Btw I'm not sure how important it is to ensure models are printable, at least in java we didn't hear any such ask, users usually access properties using `getters` and same getters used to form string to log.

I assume you mean over the wire  the property is  named`groupId` but in the model we want it to map to a property with the name `threadId`, if that's the case could we use the name attribute of JasonProperty as in, does that solves these categories of issue?

```java
public abstract class BaseEvent {
    @JsonProperty(value = "groupId")
    private String threadId;

   public String getThreadId() {
      return this.threadId;
   }
}
```

Reply:
If there's no requirement for consistency in property names in printed JSON v.s. class field name, above solution 

    @JsonProperty(value = "groupId")
    private String threadId;

is good enough for our scenario. 
Thanks! I'll apply the code change.

Reply:
Cool, yes, no such requirement; Even in AutoRest, we allow customization to rename getters, where the wire-name is considered more as impl details.